config/config_test.go

```diff
@@ -1,8 +1,135 @@
 package config
 
-import "testing"
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path"
+	"strings"
+	"testing"
 
-func TestInit(t *testing.T) {
-	// Intentionally left blank. This function exists to
-	// enforce execution of the module's init() code
+	"github.com/spf13/viper"
+)
+
+func tempDir() string {
+	dir, _ := ioutil.TempDir("", ProgramName)
+	return dir
+}
+
+// Test_Initialize_Empty tests the case where a config file and its directory
+// do not yet exist
+func Test_Initialize_Empty(t *testing.T) {
+	defer viper.Reset()
+	dir := tempDir()
+	defer os.RemoveAll(dir)
+	// additional non-existing sub directory
+	dir = path.Join(dir, "subdir")
+
+	err := Initialize(dir)
+	if err != nil {
+		t.Error("Error in Initialize:", err)
+	}
+
+	Config.Email = "email@example.com"
+	Config.Token = "some-token"
+
+	err = WriteToFile()
+	if err != nil {
+		t.Error(err)
+	}
+	content, readErr := ioutil.ReadFile(ConfigFilePath)
+	if readErr != nil {
+		t.Error(readErr)
+	}
+	yamlText := string(content)
+	if !strings.Contains(yamlText, "updated:") {
+		t.Log(yamlText)
+		t.Error("Written YAML doesn't contain the expected string 'updated:'")
+	}
+}
+
+// Test_Initialize_NonEmpty tests initializing with a dummy config file
+func Test_Initialize_NonEmpty(t *testing.T) {
+	defer viper.Reset()
+	dir := tempDir()
+	defer os.RemoveAll(dir)
+	filePath := path.Join(dir, ConfigFileName+"."+ConfigFileType)
+
+	// write dummy config
+	file, fileErr := os.Create(filePath)
+	if fileErr != nil {
+		t.Error(fileErr)
+	}
+	file.WriteString("email: email@example.com\ntoken: some-token\n")
```

Comment:
The tests are great but I'd prefer it the email and token were vars as they are used multiple times.